sys/include/xtimer.h

```diff
@@ -279,6 +280,19 @@ int xtimer_msg_receive_timeout(msg_t *msg, uint32_t us);
 int xtimer_msg_receive_timeout64(msg_t *msg, uint64_t us);
 
 /**
+ * @brief lock a mutex but with timeout
+ *
+ * @note this requires core_thread_flags to be enabled
+ *
+ * @param[in]    mutex  mutex to lock
+ * @param[in]    us     timeout in microseconds relative
+ *
+ * @return       0, when returned after mutex was locked
+ * @return       -1, when the timeout occcured
+ */
+int xtimer_mutex_lock_timeout(mutex_t *mutex, uint64_t us);
```

Comment:
Yes, but from a review perspective it makes even more sense. The know-how about semaphores and xtimer can be assumed to be split among the reviewers, so this PR would essentially need to ACKs: one from a person knowing about xtimer and one about semaphores ;-).